darshan-util/pydarshan/darshan/experimental/plots/data_access_by_filesystem.py

```diff
@@ -0,0 +1,591 @@
+
+    file_wr_series: a ``pd.Series`` object with per-category
+    counts of unique files to which at least 1 byte has been
+    written
+
+    bytes_rd_series: a ``pd.Series`` object with per-category
+    counts of bytes read
+
+    bytes_wr_series: a ``pd.Series`` object with per-category
+    counts of bytes written
+
+    filesystem_roots: a sequence of strings containing unique filesystem root paths
+    """
+    list_byte_axes: list = []
+    list_count_axes: list = []
+    # use log10 scale if range exceeds
+    # two orders of magnitude in a column
+    use_log = [False, False]
+    for idx, series_pair in enumerate([[bytes_rd_series, bytes_wr_series],
+                                       [file_rd_series, file_wr_series]]):
+        maxval = max(series_pair[0].max(), series_pair[1].max())
+        minval = min(series_pair[0].min(), series_pair[1].min())
+        ratio = maxval / minval
+        if ratio > 100:
+            use_log[idx] = True
+
+    for row, filesystem in enumerate(filesystem_roots):
+        ax_filesystem_bytes = fig.add_subplot(len(filesystem_roots),
+                                              2,
+                                              row * 2 + 1)
+        ax_filesystem_counts = fig.add_subplot(len(filesystem_roots),
+                                              2,
+                                              row * 2 + 2)
+        if row > 0:
+            ax_filesystem_bytes.get_shared_x_axes().join(ax_filesystem_bytes, list_byte_axes[row - 1])
+            if use_log[0]:
+                ax_filesystem_bytes.set_xscale('log')
+            ax_filesystem_counts.get_shared_x_axes().join(ax_filesystem_counts, list_count_axes[row - 1])
+            if use_log[1]:
+                ax_filesystem_counts.set_xscale('log')
+
+        list_byte_axes.append(ax_filesystem_bytes)
+        list_count_axes.append(ax_filesystem_counts)
+
+        # convert to MiB using the factor suggested
+        # by Google (approximate result only for now)
+        bytes_read = bytes_rd_series[filesystem]/1.049e+6
+        bytes_written = bytes_wr_series[filesystem]/1.049e+6
+        files_written = file_wr_series[filesystem]
+        files_read = file_rd_series[filesystem]
+
+        # scale to fit longer filesystem
+        # strings on the left side of the plots
+        # NOTE: may need more sophisticated scaling
+        # eventually
+        if len(filesystem) <= 8 and not '<STD' in filesystem:
+            fontsize = 18
+        else:
+            fontsize = 12
+
+        # anonymized STD.. streams have associated integers
+        # that are stored in the filesystem data field
+        # but that are confusing to display, so strip them
+        if filesystem.startswith('anonymized'):
+            ax_filesystem_bytes.annotate('anonymized', (-0.3, 0.5), fontsize=fontsize, xycoords='axes fraction')
+        else:
+            ax_filesystem_bytes.annotate(filesystem, (-0.3, 0.5), fontsize=fontsize, xycoords='axes fraction')
```

Comment:
For these `annotate()` calls we can add the `va='center'` parameter to vertically center the text on the subplots. It's not super obvious for cases with few root file systems, but for cases with 6+ it becomes noticeable and looks a little awkward. I have a few examples posted in a comment below where you can see the file system roots are not aligned.